### PR TITLE
mgr/cephadm: Show warning when user provides --fsid option

### PR DESCRIPTION
**Pull request by rkachach**
Fixes: https://tracker.ceph.com/issues/50804

Signed-off-by: Redouane Kachach <rkachach@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by mgfritch — COMMENTED** [review]
maybe this should be an error if the specified `fsid` already exists?

**Review by phlogistonjohn — APPROVED** [review]
lgtm

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -5015,6 +5015,13 @@ def command_bootstrap(ctx):
     if not ctx.output_pub_ssh_key:
         ctx.output_pub_ssh_key = os.path.join(ctx.output_dir, 'ceph.pub')
 
+    if ctx.fsid:
+        data_dir_base = os.path.join(ctx.data_dir, ctx.fsid)
+        if os.path.exists(data_dir_base):
+            raise Error(f"A cluster with the same fsid '{ctx.fsid}' already exists.")
+        else:
+            logger.warning('Using --fsid option may cause fsid conflicts.')
```
[the remark this answers is not in the document]
Comment (adk3798) [reply src/cephadm/cephadm:5023]: I still feel like the error message could use a bit more nuance. Something like `While supported, specifying an fsid for your cluster offers no advantages and may increase the likelihood of fsid conflicts`. The way it's written right now sounds a bit too serious imo. As if this is actually causing a big problem when in reality it is just a higher chance of an issue in the future.

[the remark this answers is not in the document]
Comment (rkachach) [reply src/cephadm/cephadm]: I updated the warning message using Adam suggestion.

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
http://pulpito.front.sepia.ceph.com/adking-2022-03-29_23:26:23-orch:cephadm-wip-adk-testing-2022-03-29-1402-distro-basic-smithi/

Failures:
https://tracker.ceph.com/issues/44587
https://tracker.ceph.com/issues/49287
1 machine failed to install kernel fast enough

**Pull request merged**